test/test_torch.py

```diff
@@ -17797,8 +17797,8 @@ def test(probs, replacement):
             test(z, True)
 
     def test_multinomial_empty(self, device):
-        probs = torch.ones(0, 3)
-        num_samples = 1
+        probs = torch.ones(0, 128, device=device)
+        num_samples = 64
```

Comment:
Curiously, test passed with
```
probs = torch.ones(0, 3, device=device)
num_samples = 1
```

Reply:
Is it smth with replacement, as it fails only when num_samples > 1?

Reply:
Possible. I haven't actually stepped through the exact kernel code where it is failing.

Reply:
Test passes with replacement=False, so you have to separately test different replacement modes.

Reply:
```python
>>> torch.multinomial(x, 1, True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: CUDA error: invalid configuration argument
```

The above error comes from 
https://github.com/pytorch/pytorch/blob/58148c85f4c7db372602ed9332fa6ea0ade3e703/aten/src/ATen/native/cuda/MultinomialKernel.cu#L320-L342

For `num_samples > 1`,

```python
>>> torch.multinomial(x, 2, True)
Floating point exception (core dumped)
```

https://github.com/pytorch/pytorch/blob/58148c85f4c7db372602ed9332fa6ea0ade3e703/aten/src/ATen/native/cuda/MultinomialKernel.cu#L364-L383

Here `grid_y` is `0` and thus in the snippet below, we get floating point exception due to Divide By Zero.
https://github.com/pytorch/pytorch/blob/58148c85f4c7db372602ed9332fa6ea0ade3e703/aten/src/ATen/native/cuda/MultinomialKernel.cu#L381

The fix takes care of both these cases.